### PR TITLE
task: added refresh proposal data call after execute proposal action

### PR DESCRIPTION
**Pull request by OnlyJousting**
Changes:
- Added api call to refresh proposal data after successful execute proposal action

### CODE REVIEW

File: apps/dao/src/store/createProposalsSlice.ts [hunk 1]
```diff
@@ -477,19 +477,50 @@ const createProposalsSlice = (set: SetState<State>, get: GetState<State>): Propo
...
+
+          // update proposal executed status
+          const response = await fetch(
+            `https://prices.curve.fi/v1/dao/proposals/details/${voteType.toLowerCase()}/${voteId}?tx_hash=${voteResponseHash}`,
+          )
+          const data: PricesProposalResponse = await response.json()
+
+          if ('detail' in data) {
+            return
+          }
```
Comment (DanielSchiavini) [inline apps/dao/src/store/createProposalsSlice.ts:492]: What does this mean? Maybe a comment would be nice

Comment (OnlyJousting) [reply apps/dao/src/store/createProposalsSlice.ts:492]: sure, when the is no data the api returns an object with only a "detail" field

File: apps/dao/src/store/createProposalsSlice.ts [hunk 2]
```diff
@@ -477,19 +477,50 @@ const createProposalsSlice = (set: SetState<State>, get: GetState<State>): Propo
...
-              error: null,
-            },
-          })
-
           dismissDeploying()
           const successNotificationMessage = t`Proposal executed successfully!`
           notifyNotification(successNotificationMessage, 'success', 15000)
+
+          // update proposal executed status
+          const response = await fetch(
```
Comment (0xAlunara) [inline apps/dao/src/store/createProposalsSlice.ts:488]: Can't you just call `getProposal()` again to fetch fresh data? It feels as if this is code duplication at this point. Alternatively, just add an optional `tx_hash` parameter to `getProposal()`.

**Review by 0xAlunara — CHANGES_REQUESTED** [review]

File: apps/dao/src/store/createProposalsSlice.ts (hunk 2 above)
Comment (OnlyJousting) [reply apps/dao/src/store/createProposalsSlice.ts:488]: I could, I don't want to initiate a loading state so this runs in the background without triggering any spinners but that can also be optionalised ofc

Comment (OnlyJousting) [reply apps/dao/src/store/createProposalsSlice.ts:488]: Yea, this is a good suggestion. I wanted to do a silent refetch of data but can do it within the getProposal method.

**Review by 0xAlunara — DISMISSED** [review]

File: apps/dao/src/store/createProposalsSlice.ts [hunk 3]
```diff
@@ -158,17 +158,18 @@ const createProposalsSlice = (set: SetState<State>, get: GetState<State>): Propo
...
-    getProposal: async (voteId: number, voteType: ProposalType, silentFetch = false) => {
+    getProposal: async (voteId: number, voteType: ProposalType, silentFetch = false, txHash?: string) => {
       if (!silentFetch) {
         get()[sliceKey].setStateByKey('proposalLoadingState', 'LOADING')
       }
 
       try {
         const proposal = await fetch(
-          `https://prices.curve.fi/v1/dao/proposals/details/${voteType.toLowerCase()}/${voteId}`,
+          `https://prices.curve.fi/v1/dao/proposals/details/${voteType.toLowerCase()}/${voteId}${!!txHash ? `?tx_hash=${txHash}` : ''}`,
```
Comment (DanielSchiavini) [inline apps/dao/src/store/createProposalsSlice.ts:168]:
```suggestion
          `https://prices.curve.fi/v1/dao/proposals/details/${voteType.toLowerCase()}/${voteId}${txHash ? `?tx_hash=${txHash}` : ''}`,
```

**Review by DanielSchiavini — APPROVED** [review]

**Review by 0xAlunara — APPROVED** [review]

**Pull request merged**